### PR TITLE
26 email sending service

### PR DESCRIPTION
**Pull request by bochka123**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -395,4 +395,7 @@ FodyWeavers.xsd
 *.msp
 
 # JetBrains Rider
-*.sln.iml
\ No newline at end of file
+*.sln.iml
+
+**/FirebaseServiceAccountKey.json
+**/EasySpeak.Emailer.WebAPI/appsettings.json
```
Comment (Anatolii-Ivanov) [inline .gitignore:401]: Please, right click on project name, "User Secrets", add hidden items there, don't exclude appsettings please
Or use appsettings.Development.json

File: README.md [hunk 2]
```diff
@@ -29,6 +29,7 @@ By default, apps run on the following ports:
 | EasySpeak.**Core**     | 5050 |
 | EasySpeak.**Notifier** | 5070 |
 | RabbitMQ               | 5672 |
+| EasySpeak.**Emailer**  | 7109 |
```
Comment (Anatolii-Ivanov) [inline README.md:32]: Let's go with port 5000 something? Like 5090

File: backend/EasySpeak.Core/EasySpeak.Core.WebAPI/Controllers/EmailerController.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+            _httpClient = new HttpClient();
+            _httpClient.BaseAddress = new Uri("https://localhost:7109/api/emailer/send");
+        }
+
+        [HttpPost("Send")]
+        public async Task<ActionResult<NewMailDto>> SendEmail(NewMailDto mail)
+        {
+            var request = await _httpClient.PostAsync("", new StringContent(JsonConvert.SerializeObject(mail), Encoding.UTF8, "application/json"));
+            return StatusCode((int)request.StatusCode, request);
+        }
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Core/EasySpeak.Core.WebAPI/Controllers/EmailerController.cs:25]: Can you move this logic to a separate service, so we can reuse it from code? As we won't send mails by HTTP requests

File: backend/EasySpeak.Emailer/EasySpeak.Emailer.WebAPI/Program.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+app.MapPost("/api/emailer/send", async (IValidator<NewMailDto> validator, IMailService mailService, NewMailDto mailDto) =>
+{
+    ValidationResult validationResult = await validator.ValidateAsync(mailDto);
+    if (!validationResult.IsValid)
+    {
+        return Results.ValidationProblem(validationResult.ToDictionary());
+    }
+    await mailService.SendEmailAsync(mailDto.To, mailDto.Subject, mailDto.Content);
+    return Results.Created($"/{mailDto.To}", mailDto);
+});
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Emailer/EasySpeak.Emailer.WebAPI/Program.cs:43]: Can you move this function to a separate file async (IValidator<NewMailDto> validator, IMailService mailService, NewMailDto mailDto) =>...
And then use it here

File: backend/EasySpeak.Emailer/EasySpeak.Emailer.WebAPI/Properties/launchSettings.json [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  },
+  "profiles": {
+    "EasySpek.Emailer": {
+      "commandName": "Project",
+      "launchBrowser": true,
+      "launchUrl": "swagger",
+      "environmentVariables": {
+        "ASPNETCORE_ENVIRONMENT": "Development"
+      },
+      "applicationUrl": "https://localhost:7109;http://localhost:5109",
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Emailer/EasySpeak.Emailer.WebAPI/Properties/launchSettings.json:19]: You can change project port here

File: backend/EasySpeak.Core/EasySpeak.Core.WebAPI/Controllers/EmailerController.cs [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    [Route("api/[controller]")]
+    [ApiController]
+    public class EmailerController : ControllerBase
+    {
+        private HttpClient _httpClient;
+
+        public EmailerController()
+        {
+            _httpClient = new HttpClient();
+            _httpClient.BaseAddress = new Uri("https://localhost:7109/api/emailer/send");
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Core/EasySpeak.Core.WebAPI/Controllers/EmailerController.cs:17]: Move to appsettings

File: backend/EasySpeak.Emailer/EasySpeak.Emailer.WebAPI/Services/SendGridMailService.cs [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+        private IConfiguration _configuration;
+        public SendGridMailService(IConfiguration configuration)
+        {
+            _configuration = configuration;
+        }
+        public async Task SendEmailAsync(string toEmail, string subject, string content)
+        {
+            var apiKey = _configuration["SendGridAPIKey"];
+            var client = new SendGridClient(apiKey);
+            var from = new EmailAddress("easymeats.service@gmail.com", "EasySpeak");
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Emailer/EasySpeak.Emailer.WebAPI/Services/SendGridMailService.cs:18]: Email -> appsettigns

File: backend/EasySpeak.Notifier/EasySpeak.Notifier.WebAPI/appsettings.json
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Notifier/EasySpeak.Notifier.WebAPI/appsettings.json:1]: Leave it, please

File: docker-compose.apps.prod.yml [hunk 8]
```diff
@@ -16,6 +16,19 @@ services:
         volumes:
            - ./resources:/app/Resources
 
+    easyspeak_emailer:
+        image: bochka123/easyspeak_emailer:latest
```
Comment (Anatolii-Ivanov) [inline docker-compose.apps.prod.yml:20]: Change to tatihutii/easyspeak_emailer, it may be changed and tania's account is more persistent here)

**Review by Anatolii-Ivanov — CHANGES_REQUESTED** [review]

File: backend/EasySpeak.Notifier/EasySpeak.Notifier.WebAPI/appsettings.json
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Notifier/EasySpeak.Notifier.WebAPI/appsettings.json:1]: Get it back

**Review by Anatolii-Ivanov — CHANGES_REQUESTED** [review]

File: backend/EasySpeak.Notifier/EasySpeak.Notifier.WebAPI/appsettings.json
Comment (bochka123) [reply backend/EasySpeak.Notifier/EasySpeak.Notifier.WebAPI/appsettings.json:1]: But it is there

**Review by Anatolii-Ivanov — APPROVED** [review]

**Review by potapchuk-anna — APPROVED** [review]

**Pull request merged**